Fix: Attempt to prevent sniffer from closing too fast after starting.

Skijearz opened the pull request:
This should fix https://github.com/GuilhermeFaga/hero-siege-stats/issues/20 as it checks for the `stop_cb` attribute on the sniffer object.

Skijearz committed:
fix: attempt to fix the error that the sniffer tries to stop too quick after being created and crash because not being setup all the way.

keoy7am commented:
I think it can be merged directly!
Based on the error message in #20 , this seems like a solution.
If the other party encounters other problems when using gpn
Let him re-initiate the issue

Skijearz committed:
fix: changed account network message to new version and added all missing fields - also deleted unused fields.
- also added a field for the current season_mode to support gold changes hopefully on all season_modes and characters :D

Skijearz committed:
change: gold wasnt properly updated on all chars because they were in different "seasons" as the gold message differentiates if the char is in the current season or plays hardcore etc.

Skijearz committed:
feat-request: shall we stop the app from running if it is not started on a windows machine ?

Skijearz committed:
change: added better handling of the restarting sniffer and cleaning the packet buffers because there were some errors if the restart happened right in between multiple packet reads.

Skijearz committed:
fix: implemented multiple endpoint to check for connectivity. Also prevent App from starting if no interface could be found or the running OS is not Windows. Version Checks now doesnt prevent the app from starting at all

Skijearz commented:
Hey there :D 
i have been on a hiatus for quite some time now - but i have changed, added and fixed some things.
1. In regards to #20 i have added multiple hosts to check for connection because google is not accessible for everyone - feel free to add more.
2. also in regards to #20 If the version check is not successful because github is not accessible or something like that - the app now continues to run and wont close immediately.
3. Changed the account.py network message to proper align with the actual network message.
4. Changed the Gold update behavior to support all different character modes a character can have (season,hardcord,non-season,bloodpact..)
5. changed some logic in the sniffer manager and message_parser to make it more robust with less errors
6. added the possibility to stop the app on startup if the running machine is not Windows - commented it out for now but i think thats reasonable as we use windows exclusive features. But might need feedback on that


would like some feedback from you guys @GuilhermeFaga @keoy7am  if you have the time ofc xD

Skijearz committed:
fix: add season_mode as a class attribute to default to none as it can happen that there is a gold event before the account event therefore season mode wouldnt not exist prior to that
